kroxylicious-api/src/main/java/io/kroxylicious/proxy/plugin/PluginConfigType.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.plugin;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * An annotation, on a plugin implementation class, that identifies the class of "config record"
+ * consumed by that implementation.
+ * Use {@code @PluginConfigType(Void.class)} if a plugin implementation class doesn't require configuration.
+ * @see io.kroxylicious.proxy.plugin
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+public @interface PluginConfigType {
```

Comment:
The name `PluginConfigType` makes me want to put the annotation on the config class itself.  I think this annotation would be better called `@PluginFactory` an have a formal parameter `configType`.

Reply:
I don't think `@PluginFactory` is quite right. There's nothing in this plugin mechanism that is really specific to factories. While a plugin can be a factory, it doesn't have to be. The contract for what a plugin does is independent of any of the annotations. It just happens that right now the only way this is exposed to people using the API is through the `FilterFactoryContext` which _is_ concerned with a filter factory, and so the nested plugins those developers might need the services of are also likely to be factories. 

Perhaps a better name would be something like `@ConsumesConfig` or `@RequiresConfig`?

Reply:
> @ConsumesConfig or @RequiresConfig

Those names are better. That deals with  the problem of the developer wanting to add the annotation to the config class.

However, as a developer who hadn't read the docs, the name would lead me think that the annotation is optional.  If I were writing a plugin that didn't need config, my instinct would be to omit it.

I take your point about `@PluginFactory`.   

So, how about plain old `@Plugin` which takes `configType` as a parameter?